### PR TITLE
add oak staging for genesis state/wasm

### PR DESCRIPTION
**Pull request by justinzhou93**

### CODE REVIEW

File: node/src/chain_spec/oak.rs [hunk 1]
```diff
@@ -13,8 +13,11 @@ use primitives::{AccountId, AuraId, Balance};
 
 const TOKEN_SYMBOL: &str = "OAK";
 const SS_58_FORMAT: u32 = 51;
+const STAGING_SS_58_FORMAT: u32 = 52;
```
Comment (justinzhou93) [inline node/src/chain_spec/oak.rs:16]: i know we need to register our ss58 somewhere for this to work. but we need something different from 51 i think?

File: node/src/chain_spec/oak.rs [hunk 2]
```diff
@@ -13,8 +13,11 @@ use primitives::{AccountId, AuraId, Balance};
...
 static RELAY_CHAIN: &str = "rococo-local";
+static STAGING_RELAY_CHAIN: &str = "rococo";
 const REGISTERED_PARA_ID: u32 = 2114;
+const REGISTERED_STAGING_PARA_ID: u32 = 2115;
```
Comment (justinzhou93) [inline node/src/chain_spec/oak.rs:20]: this paraID is open, but i think we need santiago to claim it before we can actually use

File: node/src/chain_spec/oak.rs [hunk 3]
```diff
@@ -89,6 +92,81 @@ pub fn oak_development_config() -> ChainSpec {
...
+				ALLOC_TOKENS_TOTAL,
+				EXISTENTIAL_DEPOSIT,
+			);
+
+			testnet_genesis(
+				// initial collators.
+				vec![
+					(
+						// SS58 prefix substrate: 5EvKT4iWQ5gDnRhkS3d254RVCfaQJSyXHj5mA4kQTsCjWchW
+						hex!["7e4f4efde71551c83714fb3062724067e7bb6ebc3bf942813321f1583e187572"]
```
Comment (justinzhou93) [inline node/src/chain_spec/oak.rs:124]: all of these invulns, sudo, council, etc are all from turing staging. do we want different accounts?

**Review by justinzhou93 — COMMENTED** [review]
generated genesis state/wasm off oak-dev, only to realize we can't use that 😐 . so couple of questions for @irsal :
1. ~are we going to be actually launching/maintaining an oak-staging on rococo or is this just for testing crowdloan?~
2. ~if we are going to maintaining an oak-staging, i had couple of questions on accounts/params~
3. ~need some ROC to test in Rococo - right now can't test contribute flow cause no ROC~
4. need the oak_staging_alloc.json to be updated with the correct values before genesis state is generated

File: node/src/chain_spec/oak.rs (hunk 1 above)
Comment (irsal) [reply node/src/chain_spec/oak.rs:16]: nope, we can use 51

Comment (irsal) [reply node/src/chain_spec/oak.rs:16]: 51 is an address format. We'd like the address format to be the same across our chains: staging, turing, oak

Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:16]: even if turing staging with ss58 51 already exists in rococo?

Comment (irsal) [reply node/src/chain_spec/oak.rs:16]: it's not. For example, mangatax uses the substrate default ss58 address. So do many of the parachains

Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:16]: awesome, saves a bunch of work. will use the same 51 ss58!

**Comment by justinzhou93** [comment]
image
image

File: node/src/chain_spec/oak.rs (hunk 3 above)
Comment (irsal) [reply node/src/chain_spec/oak.rs:123]: We do; not blocking this pr. I will change those after I gathered all wallets

**Review by irsal — CHANGES_REQUESTED** [review]
let's just do oak (no staging)

File: node/src/chain_spec/oak.rs [hunk 4]
```diff
@@ -89,6 +95,77 @@ pub fn oak_development_config() -> ChainSpec {
...
+pub fn oak_staging() -> ChainSpec {
+	// Give your base currency a unit name and decimal places
+	let mut properties = sc_chain_spec::Properties::new();
+	properties.insert("tokenSymbol".into(), TOKEN_SYMBOL.into());
+	properties.insert("tokenDecimals".into(), TOKEN_DECIMALS.into());
+	properties.insert("ss58Format".into(), SS_58_FORMAT.into());
+
+	ChainSpec::from_genesis(
+		// Name
+		"Oak Staging",
```
[the remark this answers is not in the document]
Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:107]: we want it to be called `OAK` for the staging instance too? this is the one on rococo

File: node/src/chain_spec/oak.rs (hunk 3 above)
Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:123]: cool, sounds good - just fyi, can't do genesis things for staging until this is in tho!

Comment (irsal) [reply node/src/chain_spec/oak.rs:123]: sudo should be `5C571x5GLRQwfA3aRtVcxZzD7JnzNb3JbtZEvJWfQozWE54K`

Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:123]: yes, i believe it should be! double check me, but i'm pretty sure that's what it's set as! (it's the same as turing staging)

**Comment by justinzhou93** [comment]
image
vesting looks good - timestamps are in the past, cause i just copy pasted from turing staging, but should match.

only thing looks like the timestamps aren't in the right order, but it does look like this is the case in turing as well.

File: distribution/oak_staging_alloc.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+[
```
[the remark this answers is not in the document]
Comment (justinzhou93) [reply distribution/oak_staging_alloc.json:1]: done!

File: node/src/chain_spec/oak.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (justinzhou93) [reply node/src/chain_spec/oak.rs:109]: chatted on slack - keeping oak-staging

**Review by irsal — APPROVED** [review]

**Pull request merged**